### ISSUE TITLE
match_handler always returns 0

### ISSUE THREAD

**Issue by sixers** [darvid/python-hyperscan#4]
```c
static int match_handler(unsigned int id, unsigned long long from,
                         unsigned long long to, unsigned int flags,
                         void *context) {
  ...
  return 0;
}
```

I'm new to Hyperscan, but according to the documentation [here](https://intel.github.io/hyperscan/dev-reference/runtime.html):

> The match callback function has the capability to halt scanning by returning a non-zero value.

Maybe `match_handler` should return whatever Python callback returns? This way we could also halt scanning from Python.

**Comment by darvid** [comment]
Can you try 98236220135a67c1ab494e4821638bf79cbe9658? This is a backwards incompatible change that will require all handlers to return something truthy, otherwise Hyperscan will halt stream scanning. So return `False` to halt, `True` to continue, for example. I think the negated return value allows for more Pythonic code, especially since Hyperscan doesn't care (for now) what the exact return value from Python is.

**Comment by sixers** [comment]
Hey, thank you for such a timely response!

However, I'm not sure if reversing the logic from the original hyperscan library is a good idea. It might lead to non-obvious bugs, especially when someone forgets to add a return in their function:


```
def my_handler(*args, **kwargs):
    print(*args)
```

If I understand correctly, this will halt after the first match, which might be suprising to the users. Because this package lacks documentation, people will refer to the original hyperscan docs, which won't help them in that case.

WDYT?

**Comment by darvid** [comment]
Good point, I think that makes more sense for now. It weirds me out because ``True`` and success in Python don't generally map to ``0``, and the C API expects ``1`` (failure state) from the match handler to actually halt scanning. So if you somehow had to return a bool in the C API, you would return ``False`` to stop scanning.

But I think you're right about ease of use. This is implemented in #7, along with a substantial packaging update and some [initial documentation](https://python-hyperscan.readthedocs.io/).

Let me know if there's anything else otherwise I will close the ticket. Thanks for your feedback!
